### ISSUE TITLE
batch inference GLM-4V-9B容易在英文单词处中断

### ISSUE THREAD

**Issue by ouungj** [THUDM/GLM-4#280]
### System Info / 系統信息

cuda 12.3 transfomer 4.40.0

### Who can help? / 谁可以帮助到您？

@zRzRzRzRzRzRzR

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [X] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

运行代码：
messages = [[...],[...],...,[...], [...]] #对应八个图
inputs = tokenizer.apply_chat_template(messages,
                                       add_generation_prompt=True, tokenize=True, return_tensors="pt",
                                       return_dict=True, truncation=True, padding=True)  # chat mode
inputs = inputs.to(device)

gen_kwargs = {"max_length": 8000, "do_sample": True, "top_k": 1, 'eos_token_id': model.config.eos_token_id}
 with torch.no_grad():
        outputs = model.generate(**inputs, **gen_kwargs)
        outputs = outputs[:, inputs['input_ids'].shape[1]:]

        for k in range(len(outputs)):
              answer = tokenizer.decode(outputs[k]).split()[0]

输入case：
带明显的英文单词的图片，并且要求详细描述图像，有文字的要说出文字。

出现的问题：
输出停在单词处，如：
“这张图片展示了一个放在白色装饰性盘子上的棕色袋子，袋子上有花卉图案和“For”

### Expected behavior / 期待表现

正常整个图的描述。

**Comment by ouungj** [comment]
单图推理也出现了这种情况，而且只要是多个单词连续的就一定出问题，感觉是个bug。
单图的代码：
inputs = tokenizer.apply_chat_template([{"role": "user", "image": image, "content": question_text}],
                                       add_generation_prompt=True, tokenize=True, return_tensors="pt",
                                       return_dict=True)  # chat mode
inputs = inputs.to(device)

gen_kwargs = {"max_length": 2500, "do_sample": True, "top_k": 1}
 with torch.no_grad():
        outputs = model.generate(**inputs, **gen_kwargs)
        outputs = outputs[:, inputs['input_ids'].shape[1]:]
        answer = tokenizer.decode(outputs[0]).split()[0]

**Comment by zRzRzRzRzRzRzR** [comment]
<img width="1448" alt="image" src="https://github.com/THUDM/GLM-4/assets/93239683/36aaaeeb-6147-42ad-a452-b05195616b85">
似乎没有啊，我这个全是英语单词，复现不了，单图

**Comment by ouungj** [comment]
> <img alt="image" width="1448" src="https://private-user-images.githubusercontent.com/93239683/345764706-36aaaeeb-6147-42ad-a452-b05195616b85.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwODM5NjYsIm5iZiI6MTcyMDA4MzY2NiwicGF0aCI6Ii85MzIzOTY4My8zNDU3NjQ3MDYtMzZhYWFlZWItNjE0Ny00MmFkLWE0NTItYjA1MTk1NjE2Yjg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA0VDA5MDEwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmY2Y3N2FkMDQzNWQ5MWY2YmVjNzA4Njk0ZDBjZGYwNzUzOWI1NGE2NTU3MmVhYTQ0NzA4YjdlZjI5NzA1MDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UEHLoXI0DMj5QWMj5Mxn5GIVHQltF9Otq27_VK-PgZ4"> 似乎没有啊，我这个全是英语单词，复现不了，单图

中文试试怎么样呀

**Comment by zRzRzRzRzRzRzR** [comment]
你这图像我没办法访问，我用我的图中文也可以，我是A100
